Rakefile

```diff
@@ -165,7 +165,17 @@ def _flatten_project
 
     puts "Working on #{Dir[dir + '/*.tf']}" if debug
     system("terraform get #{dir}")
-    FileUtils.cp( Dir["#{dir}/*.tf"], tmp_dir)
+    FileUtils.cp(Dir["#{dir}/*.tf"], tmp_dir)
+
+    # this has a small chance of overwriting a file if both resources/ and resources/$env
+    # have the same file
+    ['files', 'templates'].each do |subdir|
+      if File.directory? "#{dir}/#{subdir}/"
```

Comment:
Sorry, seem to have left this go stale. I believe this approach is correct and the one I stated [here](https://github.com/alphagov/govuk-terraform-provisioning/pull/46#issuecomment-213479977) is incorrect as we might want to have different files or templates with a different environment, which this loop takes account of.